src/merkle.rs

```diff
@@ -60,11 +61,12 @@ pub const BUILD_DATA_BLOCK_SIZE: usize = 64 * BUILD_CHUNK_NODES;
 ///
 /// TODO: Ord
 #[derive(Debug, Clone, Eq, PartialEq, Default)]
-pub struct MerkleTree<T, A, K>
+pub struct MerkleTree<T, A, K, U>
```

Comment:
I think you can set `U = U2` as a default parameter, this way existing implementations and usages don't have to change